diesel_codegen/src/schema_inference/sqlite.rs

```diff
@@ -0,0 +1,76 @@
+use diesel::*;
+use diesel::sqlite::SqliteConnection;
+use syntax::ast;
+use syntax::codemap::Span;
+use syntax::ext::base::*;
+use syntax::ptr::P;
+
+use super::data_structures::*;
+
+pub type InferConnection = SqliteConnection;
+
+table!{
+    pragma_table_info (cid){
+        cid ->Integer,
+        name -> VarChar,
+        type_name -> VarChar,
+        notnull -> Integer,
+        dflt_value -> Nullable<VarChar>,
+        pk -> Integer,
+    }
+}
+
+pub fn establish_connection(
+    cx: &mut ExtCtxt,
+    sp: Span,
+    database_url: &str,
+) -> Result<SqliteConnection, Box<MacResult>>
+{
+    SqliteConnection::establish(database_url).map_err(|_| {
+        cx.span_err(sp, "failed to establish a database connection");
+        DummyResult::any(sp)
+    })
+}
+
+pub fn get_table_data(conn: &SqliteConnection, table_name: &str) -> QueryResult<Vec<ColumnInformation>>
+{
+    conn.execute_pragma::<pragma_table_info::SqlType, ColumnInformation>(
+        &format!("PRAGMA TABLE_INFO('{}')", table_name))
+}
+
+pub fn determine_column_type(cx: &mut ExtCtxt, attr: &ColumnInformation) -> P<ast::Ty>
+{
+    let type_name=attr.type_name.to_lowercase();
+    //see https://www.sqlite.org/datatype3.html
+    let tpe = if type_name.contains("int"){
+        quote_ty!(cx, ::diesel::types::Int8)
+    }else if type_name.contains("char")||
+        type_name.contains("clob")||
+        type_name.contains("text")
+    {
+        quote_ty!(cx, ::diesel::types::Text)
+    }else if type_name.contains("blob") || type_name.is_empty(){
```

Comment:
The style of this whole section is really wonky. Please add spaces around your braces, and avoid multiline predicates. If something is too long to fit on a line it should be moved to a function.